blockdevice_test.go

```diff
@@ -89,7 +95,13 @@ var blockdevicesResponse = `
                 "size": 8581545984
             }
         ],
-        "filesystem": null,
+        "filesystem": {
```

Comment:
We should retain this "WithNulls" test and add a new one without

Reply:
This diff is confusing I still have the filesystem is nil in the WithNulls. I added another check to ensure that filesystem stays nil in the WithNulls.